### PR TITLE
stardict path override

### LINKED ISSUE #5995: Make dictionaries folder location configurable
* KOReader version: 2020.02 (Android)
* Device: Onyx Boox Note

#### Issue

Enhancement proposal:

koreader uses its own private location for dictionaries at koreader/data/dict as common storage for external stardict dictionaries. If users would be possible to reconfigure this storage location it would be able to share an already existing location between multiple reader applications. On the Onyx Boox devices there is typically an upper level folder named dicts used for storing stardict dictionaries for its internal applications. If users would be able to simply point koreader's dictionary location to that same location there would be no need anymore to maintain duplicate dictionary files on the device and save considerable storage capacity.

### PR DESCRIPTION
**Pull request by pazos**
Fixes #5995

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/5999)

### CODE REVIEW

File: defaults.lua [hunk 1]
```diff
@@ -251,6 +251,10 @@ NETWORK_PROXY = nil
 -- Use turbo library to handle async HTTP request
 DUSE_TURBO_LIB = false
 
+-- Absolute path to stardict files.
+-- By default they're stored in data/dict under dataDir.
+STARDICT_DATA_DIR = nil
```
Comment (Frenzie) [inline defaults.lua]: We should consider something like this to keep compatibility with system-wide sdcv configuration out of the box. Unless sdcv picks up on that regardless?
```suggestion
STARDICT_DATA_DIR = os.getenv("STARDICT_DATA_DIR")
```

Comment (pazos) [reply defaults.lua]: I don't know:

```bash
pazos@bowie:/k$ ./sdcv --only-data-dir -l
Dictionary's name   Word count
pazos@bowie:/k$ export STARDICT_DATA_DIR=/test/dict/
pazos@bowie:/k$ ./sdcv --only-data-dir -l
Dictionary's name   Word count
pazos@bowie:/k$ ./sdcv --only-data-dir --data-dir=/test/dict -l
Dictionary's name   Word count
Oxford English Dictionary    75753
Diccionario Español RAE    86023
Diccionario Inglés-Español    20737
```

Comment (Frenzie) [reply defaults.lua]: --only-data-dir overrides it

https://github.com/Dushistov/sdcv/blob/08461acab8e25dc3b1a596fc7146e079dcef3d64/src/sdcv.cpp#L131-L142

Comment (pazos) [reply defaults.lua]: Sadly it's not possible to commit your suggestion as it clashes with advanced settings. One way to circumvent this is

```lua
--- a/frontend/apps/reader/modules/readerdictionary.lua
+++ b/frontend/apps/reader/modules/readerdictionary.lua
@@ -101,7 +101,8 @@ end
 
 function ReaderDictionary:init()
     self.ui.menu:registerToMainMenu(self)
-    self.data_dir = os.getenv("STARDICT_DATA_DIR") or
+    self.data_dir = STARDICT_DATA_DIR or
+        os.getenv("STARDICT_DATA_DIR") or
         DataStorage:getDataDir() .. "/data/dict"
```

and leave `STARDICT_DATA_DIR = nil`

Comment (Frenzie) [reply defaults.lua]: That was my first thought actually, but I figured it might be slightly more elegant in defaults.lua.

But I suppose I was mixing logic and configuration, actually being inelegant.

tl;dr Yes please! :+1:

Comment (pazos) [reply defaults.lua:256]: done!
